### PR TITLE
[qt] Apply workaround for cocoa platform as well

### PR DESCRIPTION
**Pull request by Ferenc-**

### CODE REVIEW

**Comment by Ferenc-** [comment]
@vng Could you also check if this works as an alternative to https://github.com/organicmaps/organicmaps/pull/4957?

**Comment by biodranik** [comment]
@Ferenc- why do you think that we should not set the appropriate OpenGL version and core profile in some cases? If it's not done, then OpenGL 2.0 is selected by default.

**Comment by biodranik** [comment]
Does it mean that we can/should move all existing code from `main.cpp` to `map_widget`?

**Comment by vng** [comment]
The dev process for Mac is blocked until we decide what is a _workaround_ and what is a _normal_ code here.
- I don't see a problem setting OGL version at main.cpp since it is *needed* for 2 platforms already, hence this is not a  workaround already.
- I don't see a problem with fallback on OGL v2 in widget code when we have more context info.
- I don't understand why it may "distracts the code reader"
- I don't see a problem to have _double_ initialization if we can avoid any temporary resource allocation.

@Ferenc- Please, confirm that on your device fallback to OGL v2 (where it is needed) works properly. And I'm gonna merge https://github.com/organicmaps/organicmaps/pull/4957

**Comment by Ferenc-** [comment]
> The dev process for Mac is blocked until we decide what is a _workaround_ and what is a _normal_ code here.
> 
> * I don't see a problem setting OGL version at main.cpp since it is _needed_ for 2 platforms, hence this is not a  workaround already.
> * I don't see a problem with fallback on OGL v2 in widget code when we have more context info.
> * I don't understand why it may "distracts the code reader"
> * I don't see a problem to have _double_ initialization if we can avoid any temporary resource allocation.
> 
> @Ferenc- Please, confirm that on your device fallback to OGL v2 (where it is needed) works properly. And I'm gonna merge #4957

Commented in https://github.com/organicmaps/organicmaps/pull/4957#issuecomment-1509235148 that it seems to break functionality.

**Comment by vng** [comment]
Hm .. Ok, merging your branch then.
@biodranik WDYT?

**Review by vng — APPROVED** [review]

File: qt/main.cpp [hunk 1]
```diff
@@ -214,17 +214,21 @@ int main(int argc, char * argv[])
...
+    // This is a workaround for some systems,
+    // including MacOs and older Linux distros, which rely on X.org and Mesa.
+    // Where somehow the driver itself doesn't
     // make all the otherwise supported GLSL versions available by default
     // and such requests are somehow disregarded at later stages of execution.
     // This setting here will be potentially overwritten and overruled anyway,
     // and only needed to ensure that we have the needed GLSL compiler available
     // later when we need it.
-    if (app.platformName() == QString("xcb"))
+    if (app.platformName() == QString("xcb") ||
```
Comment (biodranik) [inline qt/main.cpp:226]: Why explicitly selecting Opengl ES compatibility profile and version for other platforms is bad idea?
```
android
    cocoa is a platform plugin for macOS.
    directfb
    eglfs is a platform plugin for running Qt5 applications on top of EGL and OpenGL ES 2.0 without an actual windowing system (like X11 or Wayland). For more information, see [EGLFS](https://doc.qt.io/qt-6/embedded-linux.html#eglfs).
    ios (also used for tvOS)
    linuxfb writes directly to the framebuffer. For more information, see [LinuxFB](https://doc.qt.io/qt-6/embedded-linux.html#linuxfb).
    minimal is provided as an examples for developers who want to write their own platform plugins. However, you can use the plugin to run GUI applications in environments without a GUI, such as servers.
    minimalegl is an example plugin.
    offscreen
    qnx
    windows
    wayland is a platform plugin for the Wayland display server protocol, used on some Linux desktops and embedded systems.
    xcb is a plugin for the X11 window system, used on some desktop Linux platforms.
```
?

**Pull request merged**